pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2532,17 +2533,43 @@ func handlePhysicalIOAdapterListCreateModify(ctxArg interface{},
 	ctx := ctxArg.(*domainContext)
 	phyIOAdapterList := cast.CastPhysicalIOAdapterList(configArg)
 	aa := ctx.assignableAdapters
-	log.Infof("handlePhysicalIOAdapterListCreateModify: %+v\n",
-		phyIOAdapterList)
+	log.Infof("handlePhysicalIOAdapterListCreateModify: current len %d, update %+v\n",
+		len(aa.IoBundleList), phyIOAdapterList)
+
+	if !aa.Initialized {
```

Comment:
Why do we need to special case this? if aa was never initialized - all the adapters would go into "handleIBCreate" anyways..  And by the end of this loop - we publish aa adapters with initialize set.. the fact that we are setting it in later iterations is a no-op anyways..

Reply:
Issue is that when we call handleIBCreate for e.g., Audio the code needs to be able to lookup the "Audio" bundle in AssignableAdapters, but the Audio has yet to be added.
And in general, when we add N members which are part of the same AssignmentGroup we need to set them consistently, but when member N is added only the first N-1 are present in AssignableAdapters.
Hence we need two passes; either by doing the add to the list then walk to do handleIBCreate as in this code, or if aa.Initialized was not set, call checkAndSetIoBundleAll at the end to re-evaluate everything. The latter causes a bit more churn since an adapter might be assigned to pciback first only to have the checkAndSetIoBundleAll realize that it is part of a bundle which has isPort set.